slip.js

```diff
@@ -454,21 +454,25 @@ window['Slip'] = (function(){
 
                     onEnd: function() {
                         var move = this.getTotalMovement();
+                        var i, spliceIndex;
```

Comment:
Rather than dispatch within a for-loop, and then break, the new code calculates the spliceIndex using a for-loop, and then dispatches the event.  This handles the case where the whole loop is traversed, and no otherNode is found that meets the condition of `otherNode.pos > move.y` (or `otherNode.pos < move.y` when moving the other direction).  This occurs when dragging the first or last element, and then letting go at the originalIndex.  Before this fix, the `reorder` event would only fire if you let go of the first element slightly above where it started, but not if you let go slightly below.  (And vice versa for the last element).

I also think it's a little cleaner to compute the spliceIndex separately from the dispatch code, which reduces some duplication.